docs/psa_switch-dev.md

```diff
@@ -0,0 +1,82 @@
+# Implementing Externs in PSA Switch
+Externs in PSA Switch are different than those in Simple Switch. 
+Simple Switch uses externs baked into the BMV2 model where as PSA Switch
+uses externs specific to its own architecture. To understand what this
+translates to in code, look at the counter used in 
+[Simple Switch](https://github.com/p4lang/behavioral-model/tree/master/src/bm_sim)
+and the one in 
+[PSA Switch](https://github.com/p4lang/behavioral-model/tree/master/targets/psa_switch/externs).
+The Simple Switch counter lives inside of `src/bm_sim` where as the 
+counter for PSA Switch lives inside of the target specific directory.
+This doc is meant to provide guidance for developers on how to develop
+new externs in PSA Switch.
+
+## Background Knowledge
+For those who are familiar with the internals of BMV2 feel free
+to skip this section. For those who are new/less familiar with how BMV2
+operates, consider reading `P4Objects.cpp` as a starting point
+as this is where externs will be created. From here you can also see
+what the expected JSON for externs should be. Next, read the changes
+in the following [commit](https://github.com/p4lang/behavioral-model/pull/767).
+It details all the changes made to support the counter extern in PSA.
+It will also be useful to understand on a high level how P4 Runtime
+interacts with PSA Switch. Briefly skimming `runtime_CLI.py` will do.
```

Comment:
That is incorrect, `runtime_CLI.py` does not use `P4Runtime`. `runtime_CLI.py` uses the Thrift RPC service defined in https://github.com/p4lang/behavioral-model/blob/master/thrift_src/standard.thrift. P4Runtime is a gRPC service. It is implemented over the PI C library (https://github.com/p4lang/PI) which is then itself implemented (for bmv2) in this repository (https://github.com/p4lang/behavioral-model/tree/master/PI).
Both implementations converge at `bm::RuntimeInterface` (https://github.com/p4lang/behavioral-model/blob/master/include/bm/bm_sim/runtime_interface.h).

```
runtime_CLI --- Thrift service ---> Thrift service implementation --- RuntimeInterface ---> bmv2 built-in objects + target-specific objects
```
```
[any P4Runtime client] --- P4Runtime service ---> P4Runtime implementation (DeviceMgr) --- PI API ---> PI bmv2 implementation ---> RuntimeInterface ---> bmv2 built-in objects + target-specific objects
```

"Target-specific objects" include your PSA extern implementations (you added an `override` implementation of `read_counters` and `write_counters` to `PSASwitch`).